### PR TITLE
feat: replace dependency on Bash with /bin/sh for cross-platform compatibility

### PR DESCRIPTION
**Pull request by pepa65**

### CODE REVIEW

File: main_linux.go [hunk 1]
```diff
@@ -173,7 +176,8 @@ func main() {
 				Event     string
 				Shell     string
 				Threshold int
-			}{event, shell, current}
+				Path      string
+			}{event, shell, current, first}
```
Comment (tshakalekholoane) [inline main_linux.go:180]: We can reduce the size of this struct by placing the `Threshold` field last.

**Review by tshakalekholoane — CHANGES_REQUESTED** [review]
See requested changes. Otherwise, it looks good!

File: main_linux.go [hunk 2]
```diff
@@ -69,6 +69,9 @@ func main() {
 		os.Exit(1)
 	}
 	first := batteries[0]
+	if len(batteries) > 1 {
+		fmt.Fprintln(os.Stderr, "More than 1 battery device found, using " + first)
+	}
```
[the remark this answers is not in the document]
Comment (pepa65) [reply main_linux.go:74]: Logging is probably not the best response, but if there are more than 1 battery devices, we "randomly" pick the first... We should probably select on the full path for the change limit variable, and if there is more than 1 battery, set them all.

File: README.md [hunk 3]
```diff
@@ -95,4 +95,4 @@ sudo bat persist
 
 Linux kernel version later than 5.4-rc1 which is the [earliest version to expose the battery charging threshold variable](https://github.com/torvalds/linux/commit/7973353e92ee1e7ca3b2eb361a4b7cb66c92abee).
 
-To persist the threshold setting between restarts, the application relies on [systemd](https://systemd.io/), particularly a version later than 244, and [Bash](https://www.gnu.org/software/bash/) which are bundled with most Linux distributions. 
+To persist the threshold setting between restarts, the application relies on [systemd](https://systemd.io/), particularly a version later than 244, and a POSIX shell `sh`, which are bundled with most Linux distributions.
```
[the remark this answers is not in the document]
Comment (pepa65) [reply README.md:98]: Sure, we can leave this out now.
Notice that I still query the path for `sh` as you did for `bash`, as this is what POSIX recommends. We could also just hardcode `/bin/bash` and be done with it. The way it happens now, you're likely to get `/usr/bin/sh` for Shell, which is OK, but looks a bit strange.

File: main_linux.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pepa65) [reply main_linux.go:74]: It is best to check for the full path of the threshold variable, so we really only find supported battery devices. If there are more than 1, we will want to run things on both batteries, but for the reporting we need to include the battery name `BAT?` in the output strings.

Or we can just decide that this utility is only going to handle 1 (the first found) compatible battery (which will be the majority or even all cases).

File: main_linux.go (hunk 1 above)
Comment (pepa65) [reply main_linux.go:180]: How does that change the size?

File: main_linux.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tshakalekholoane) [reply main_linux.go:74]: Yes, I don't think we should account for additional batteries if we are uncertain that such devices exist in the first place. I also don't think this information is useful for the end user so I'd be in favour of omitting it entirely.

[the remark this answers is not in the document]
Comment (pepa65) [reply main_linux.go:74]: If such a device does exist, you do want to notify the user that this application (in its current state) only handles 1 of their batteries.

In most cases, this message never shows up, but when you do find 2 batteries that expose `charge_control_end_threshold` then you do need to confess that you're not handling the other one(s).

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tshakalekholoane) [reply README.md:98]: Yes, I was also thinking about that but given the location is not guaranteed to be in `/bin/`, I think the safe thing is still to search the PATH variable.

File: main_linux.go (hunk 1 above)
Comment (tshakalekholoane) [reply main_linux.go:180]: I mean it doesn't affect the size per se, just [how the garbage collector does the mark phase](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.13.0:go/analysis/passes/fieldalignment/fieldalignment.go;drc=e5f7ae13a79f624b7414958de3f8c64693adfe8c;l=29).

File: main_linux.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tshakalekholoane) [reply main_linux.go:74]: I don't see why? I mean sure it's nice but what's the point if the information is not actionable? 

There is also a significant cost to this, it introduces a system call in the critical path and if you're running this inside something like a status bar to for example, show the current charging level, then it means users not only have to incur that cost additional cost, but they also have to filter that message out before they can read in what they actually want. 

I don't think it's worth it and in any case it's unrelated to this change. If we want to account for additional batteries, then we are better off doing it in separate change where we can also address other things that come with that.

**Review by tshakalekholoane — APPROVED** [review]

**Pull request merged**